config/initializers/elasticsearch.rb

```diff
@@ -1,23 +1,53 @@
-config = {
-  host: ENV.fetch("ES_HOST", "http://127.0.0.1:9200"),
-  transport_options: {
-    request: { timeout: 5 }
-  }
-}
+require 'base64'
+
+CONFIG_PATH = "#{Rails.root}/config/elasticsearch.yml".freeze
+TEMPLATE = ERB.new File.new(CONFIG_PATH).read
+ELASTIC_CONFIG = YAML.load(TEMPLATE.result(binding))[ENV['RAILS_ENV']]
 
-if File.exist?("config/elasticsearch.yml")
-  config.merge!(YAML.load_file("config/elasticsearch.yml")[Rails.env].symbolize_keys)
+def log(server, filepath)
+  Rails.logger.info "Configuring Elasticsearch on PAAS.\n
+  Elasticsearch host: #{server}\n
+  Elasticsearch cert file path: #{filepath}"
 end
 
-Elasticsearch::Model.client = Elasticsearch::Client.new(config)
-
-# Reset the search index before testing
-if Rails.env == "test"
-  client = ::Dataset.__elasticsearch__.client
-  begin
-    client.indices.delete index: ::Dataset.__elasticsearch__.index_name
-  rescue
-    puts "No test search index to delete"
-  end
-  client.indices.create index: ::Dataset.__elasticsearch__.index_name
+def es_config_production
+  vcap = JSON.parse(ELASTIC_CONFIG['vcap_services'])
+
```

Comment:
Boo. I get the point Code Climate but I think it's best to keep as it for readability